Bottom: b5065925836c586fe1886a7b902cf8e1bb4f0dd0
Top:    b7eecf757098a9b7991293a2470ae16319914dd4
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-12-08 15:40:07 +0800

Refresh of tuntap-rename-struct

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 885e63f..3c7da00 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -2551,7 +2551,8 @@ static int tun_set_queue(struct file *file, struct ifreq *ifr)
 	return ret;
 }
 
-static int tun_set_steering_ebpf(struct tun_struct *tun, void __user *data)
+static int tun_set_ebpf(struct tun_struct *tun, struct tun_prog **prog_p,
+			void __user *data)
 {
 	struct bpf_prog *prog;
 	int fd;
@@ -2567,7 +2568,7 @@ static int tun_set_steering_ebpf(struct tun_struct *tun, void __user *data)
 			return PTR_ERR(prog);
 	}
 
-	return __tun_set_ebpf(tun, &tun->steering_prog, prog);
+	return __tun_set_ebpf(tun, prog_p, prog);
 }
 
 static long __tun_chr_ioctl(struct file *file, unsigned int cmd,
@@ -2847,7 +2848,7 @@ static long __tun_chr_ioctl(struct file *file, unsigned int cmd,
 		break;
 
 	case TUNSETSTEERINGEBPF:
-		ret = tun_set_steering_ebpf(tun, argp);
+		ret = tun_set_ebpf(tun, &tun->steering_prog, argp);
 		break;
 
 	default:
